pkg/proxy/filter_endpoints_test.go

```diff
@@ -117,42 +117,22 @@ func TestFilterEndpoints(t *testing.T) {
 			{ip: "10.1.2.5", zoneHints: sets.NewString("zone-c")},
 			{ip: "10.1.2.6", zoneHints: sets.NewString("zone-a")},
 		},
-		expectedEndpoints: []endpoint{
-			{ip: "10.1.2.3", zoneHints: sets.NewString("zone-a")},
-			{ip: "10.1.2.4", zoneHints: sets.NewString("zone-b")},
-			{ip: "10.1.2.5", zoneHints: sets.NewString("zone-c")},
```

Comment:
Were the zoneHints in expectedEndpoints not being used then?

Reply:
They weren't doing anything _useful_; we already knew that FilterEndpoints doesn't _change_ the endpoints. But the test case was comparing the input and output with DeepEqual, so we had to re-specify all the fields again even though we knew they weren't going to change.

And this got annoying when rewriting the tests because I had to keep adding more fields to the expectedEndpoints to make them fully match the input endpoints.